admin_pages/events/Events_Admin_Page.core.php

```diff
@@ -2095,16 +2092,38 @@ private function _change_event_status($EVT_ID = 0, $event_status = '')
 
 
     /**
-     * _delete_event
-     *
-     * @access protected
-     * @param bool $redirect_after
+     * @param array $event_ids
+     * @return array
+     * @since   $VID:$
+     */
+    private function cleanEventIds(array $event_ids)
+    {
+        return array_map('absint', $event_ids);
+    }
+
+
+    /**
+     * @return array
+     * @since   $VID:$
+     */
+    private function getEventIdsFromRequest()
+    {
+        $event_ids = isset($this->_req_data['EVT_ID']) ? $this->_req_data['EVT_ID'] : [];
+        $event_ids = is_string($event_ids) ? explode(',', $event_ids) : (array) $event_ids;
+        return $this->cleanEventIds($event_ids);
+    }
+
+
+    /**
+     * @param bool $preview_delete
+     * @throws EE_Error
      */
-    protected function _delete_event()
+    protected function _delete_event($preview_delete = true)
     {
-        $this->generateDeletionPreview(isset($this->_req_data['EVT_ID']) ? $this->_req_data['EVT_ID'] : array());
+        $this->_delete_events($preview_delete);
```

Comment:
wasn't this nice and long? :trollface:

Reply:
that's partly the languages fault...
with PHP 7.1 you could do it like:

```php
$this->generateDeletionPreview($this->_req_data['EVT_ID'] ?? []);
```

which isn't too too bad